common/build/eslint-plugin-fluid/src/rules/no-unchecked-record-access.js

```diff
@@ -303,85 +313,168 @@ function isUndefinableIndexSignatureType(parserServices, node) {
-	while (current && current.parent) {
-		// Handle if statements that check property existence
-		if (current.parent.type === "IfStatement" && current.parent.test) {
-			const base = getBaseObject(node);
-			if (base && current.parent.test.type === "MemberExpression") {
-				const testBase = getBaseObject(current.parent.test);
-				if (base === testBase) {
-					return true;
-				}
-			}
-		}
-
-		// Handle for...of with Object.entries/keys
-		if (current.parent.type === "ForOfStatement") {
-			const right = current.parent.right;
+		// Handle Object.entries/keys loops
+		if (parent.type === "ForOfStatement") {
+			const right = parent.right;
 			if (
-				right.type === "CallExpression" &&
-				right.callee.type === "MemberExpression" &&
+				right?.type === "CallExpression" &&
+				right.callee?.type === "MemberExpression" &&
 				right.callee.object.name === "Object" &&
 				(right.callee.property.name === "entries" || right.callee.property.name === "keys")
 			) {
-				const base = getBaseObject(node);
-				if (right.arguments[0].type === "Identifier" && right.arguments[0].name === base) {
+				return true;
+			}
+		}
+
+		// Handle presence checks
+		if (parent.type === "IfStatement" || parent.type === "ConditionalExpression") {
+			if (parent.test) {
+				if (nodesAreEquivalent(parent.test, node)) {
 					return true;
 				}
+				if (parent.test.type === "BinaryExpression") {
+					if (parent.test.operator === "in") {
+						const testBase = getBaseObject(parent.test.right);
+						if (baseObj === testBase) {
+							// Check if the else block assigns the key
+							const ifStatement = parent;
+							const elseBlock = ifStatement.alternate;
+							if (elseBlock) {
+								const keyNode = parent.test.left;
+								if (
+									checkElseBlockAssignsKey(elseBlock, testBase, keyNode, context)
+								) {
+									return true;
+								}
+							}
+							return true;
+						}
+					}
+					if (
+						(parent.test.operator === "!==" || parent.test.operator === "!=") &&
+						((nodesAreEquivalent(parent.test.left, node) &&
+							isUndefinedNode(parent.test.right)) ||
+							(nodesAreEquivalent(parent.test.right, node) &&
+								isUndefinedNode(parent.test.left)))
+					) {
+						return true;
+					}
+				}
 			}
 		}
 
-		// Handle "in" operator checks
-		if (
-			(current.parent.type === "ConditionalExpression" ||
-				current.parent.type === "IfStatement") &&
-			current.parent.test?.type === "BinaryExpression" &&
-			current.parent.test.operator === "in"
-		) {
-			const test = current.parent.test;
-			const base = getBaseObject(node);
-			const testBase = getBaseObject(test.right);
-			if (base === testBase) {
-				return true;
+		// Additional check for 'in' checks in the same scope
+		const containingBlock = findContainingBlock(current);
+		if (containingBlock) {
+			for (const stmt of containingBlock.body) {
+				if (stmt.range && stmt.range[0] > current.range[0]) break; // Only check statements before current node
```

Comment:
Is there any need to check `stmt.length` before accessing?

Reply:
There is a reason to put `stmt.range[0] > current.range[0]`, but I didnt have to put stmt.range, it will always exist. I just found some documentation that put this so I included it. after removing it and running it, it still works